packages/client-assertion-validation/src/types.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  AgreementId,
+  ApiError,
+  ClientId,
+  clientKindTokenStates,
+  EServiceId,
+  ItemState,
+  PurposeId,
+  TenantId,
+} from "pagopa-interop-models";
+import { z } from "zod";
+import { ErrorCodes } from "./errors.js";
+
+export const ClientAssertionDigest = z
+  .object({
+    alg: z.string(),
+    value: z.string(),
+  })
+  .strict();
+export type ClientAssertionDigest = z.infer<typeof ClientAssertionDigest>;
+
+export const ClientAssertionHeader = z
+  .object({
+    kid: z.string(),
+    alg: z.string(),
+    typ: z.string().optional(),
+  })
+  .strict();
+export type ClientAssertionHeader = z.infer<typeof ClientAssertionHeader>;
+
+export const ClientAssertionPayload = z
+  .object({
+    sub: z.string(),
```

Comment:
```suggestion
    sub: z.string(),
```
Since this has to be an UUID, couldn't we declare it as `z.string().uuid()`?

Reply:
Updated